File: 02-calculations/03-circle-area/solution.c
#include <math.h>
#include <stdio.h>

float get_radius(void);

int main(void)
{
    float radius = get_radius();

    if (radius == 0.0)
    {
        printf("Invalid radius value.\n");
        return 0;
    }

    double pi = 3.141592653589793;
    double area = pi * radius * radius;
    printf("Circle area is %f\n", area);
    return 0;
}


float get_radius(void)
{
    printf("Enter the radius of the circle: ");
    float radius;

    if (scanf("%f", &radius) != 1)
    {
        return 0.0;
    }

    if (radius <= 0)
    {
        return 0.0;
    }

    return radius;
}

File: 02-calculations/03-circle-area/solution.js
import readline from 'readline/promises';
import { stdin as input, stdout as output } from 'process';

const rl=readline.createInterface({
    input, output
});


async function getRadius() {
    let radius = await rl.question("Enter the radius of the circle: ");
    radius = parseFloat(radius);

    if (isNaN(radius) || radius <= 0){
        return null;
    }
    return radius;
}


async function main() {
    const radius = await getRadius();
    if (radius === null){
        console.log("Invalid radius value.")
    }else{
        const area = Math.PI * radius * radius;
        console.log(`Circle area is ${area}`);
    }
    rl.close()
}


main()